client/cypress/e2e/add-user.cy.ts

```diff
@@ -90,13 +90,36 @@ describe('Add user', () => {
         role: 'editor'
       };
 
+      // The `page.addUser(user)` call ends with clicking the "Add User"
+      // button on the interface. That then leads to the client sending an
+      // HTTP request to server, which is has to process that request
+      // (including making calls to add the user to the database and wait
+      // for those to respond) before we get a response and can update the GUI.
+      // By calling `cy.intercept()` we're saying we want Cypress to "notice"
+      // when we go to `/api/users`. The `AddUserComponent.submitForm()` method
+      // routes to `/api/users/{MongoDB-ID}` if the REST request to add the user
+      // succeeds, and that routing will get "noticed" by the Cypress because
+      // of the `cy.intercept()` call.
+      //
+      // The `.as('addUser')` call basically gives that event a name (`addUser`)
+      // which we can use in things like `cy.wait()` to say which event or events
```

Comment:
What does it look like to wait for more than one event?

Reply:
You'd probably need to look at the docs where they have examples of this. I think an example would be a complex page that makes several requests to assemble its content (e.g., the current blog post, a list of titles of recent posts, and some ads), and you might want/need to wait until all those requests completed before you start checking the state of the page.